jota/src/test/java/org/iota/jota/IotaCoreApiTest.java

```diff
@@ -1,226 +1,298 @@
-                new String[]{TAG},
-                new String[]{test}, 
-                new String[]{test});
+        String[] addresses = {Checksum.addChecksum(TEST_BUNDLE)};
+        String[] tags = {TAG};
+        String[] approves = {TEST_BUNDLE};
+        String[] bundles = {TEST_BUNDLE};
 
-        assertNotNull(resp);
+        when(iotaAPIMock.findTransactions(addresses, tags, approves, bundles)).thenReturn(new FindTransactionResponse());
+
+        FindTransactionResponse transactionResponse = iotaAPIMock.findTransactions(addresses, tags, approves, approves);
+        assertNotNull(transactionResponse);
     }
 
     @Test
-    @Tag("IntegrationTest")
     public void findTransactionsFailIfInvalidTagIsProvided() {
         String test = TEST_BUNDLE;
+
+        when(iotaAPIMock.findTransactions(
+                new String[]{Checksum.addChecksum(test)}, new String[]{test},
+                new String[]{test}, new String[]{test}))
+                .thenThrow(new ArgumentException("Invalid tag provided."));
+
         ArgumentException argumentException = assertThrows(ArgumentException.class,
-                () -> proxy.findTransactions(
+                () -> iotaAPIMock.findTransactions(
                         new String[]{Checksum.addChecksum(test)}, new String[]{test},
                         new String[]{test}, new String[]{test}));
+
         assertEquals("Invalid tag provided.", argumentException.getMessage());
     }
 
+    @SuppressWarnings("deprecation")
     @Test
-    @Tag("IntegrationTest")
-    public void shouldGetBalances() throws ArgumentException {
-        GetBalancesResponse res = proxy.getBalances(100, Collections.singletonList(TEST_ADDRESS_WITH_CHECKSUM), null);
-        assertThat(res.getReferences(), IsNull.notNullValue());
-        assertThat(res.getBalances(), IsNull.notNullValue());
-        assertThat(res.getMilestoneIndex(), IsNull.notNullValue());
-        assertThat(res.getDuration(), IsNull.notNullValue());
+    public void shouldGetBalances() {
+        GetBalancesResponse balancesResponseMock = mock(GetBalancesResponse.class);
+
+        when(iotaAPIMock.getBalances(100, Collections.singletonList(TEST_ADDRESS_WITH_CHECKSUM), null))
+                .thenReturn(balancesResponseMock);
+
+        GetBalancesResponse balancesResponse = iotaAPIMock.getBalances(100, Collections.singletonList(TEST_ADDRESS_WITH_CHECKSUM), null);
+
+        when(balancesResponseMock.getReferences()).thenReturn(new String[]{});
+        when(balancesResponseMock.getBalances()).thenReturn(new String[]{});
+        when(balancesResponseMock.getDuration()).thenReturn(0L);
+
+        assertThat(balancesResponse.getReferences(), IsNull.notNullValue());
+        assertThat(balancesResponse.getBalances(), IsNull.notNullValue());
+        assertThat(balancesResponse.getMilestoneIndex(), IsNull.notNullValue());
+        assertThat(balancesResponse.getDuration(), IsNull.notNullValue());
```

Comment:
Codacy found an issue: [JUnit assertions should include a message](https://app.codacy.com/manual/Iota-java/iota-java/pullRequest?prid=6319461)